src/pages/files/index.tsx

```diff
@@ -38,131 +38,112 @@ export default function Files(): ReactElement {
-        const link = document.createElement('a')
-        link.href = downloadUrl
-        link.setAttribute('download', 'file.zip') //any other extension
-        document.body.appendChild(link)
-        link.click()
-        link.remove()
-      })
-      .catch(() => {
-        // FIXME: handle the error
-      })
-      .finally(() => {
-        setLoadingSearch(false)
-      })
-  }
+  const [uploadError, setUploadError] = useState<Error | null>(null)
+  const [isUploadingFile, setIsUploadingFile] = useState(false)
 
   const uploadFile = () => {
-    setUploadingFile(true)
+    if (file === null) return
+    setIsUploadingFile(true)
+    setUploadError(null)
     beeApi.files
-      .uploadFile(files[0])
+      .uploadFile(file)
       .then(hash => {
         setUploadReference(hash)
-        setFiles([])
-      })
-      .catch(() => {
-        // FIXME: handle the error
+        setFile(null)
       })
+      .catch(setUploadError)
       .finally(() => {
-        setUploadingFile(false)
+        setIsUploadingFile(false)
       })
   }
 
   const handleChange = (files?: File[]) => {
     if (files) {
-      setFiles(files)
+      setFile(files[0])
+      setUploadReference('')
     }
   }
 
+  const handleSearchChange = (e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+    setSearchInput(e.target.value)
+
+    if (Utils.Hex.isHexString(e.target.value, 64)) setSearchError(null)
+    else setSearchError(new Error('Incorrect format of swarm hash'))
+  }
+
+  if (isLoadingHealth || isLoadingNodeHealth) {
+    return (
+      <Container style={{ textAlign: 'center', padding: '50px' }}>
+        <CircularProgress />
+      </Container>
+    )
+  }
+
+  if (!health || nodeHealth?.status !== 'ok') return <TroubleshootConnectionCard />
+
   return (
-    <div>
-      {
-        // FIXME: this should be broken up
-        /* eslint-disable no-nested-ternary */
-        nodeHealth?.status === 'ok' && health ? (
-          <Container maxWidth="sm">
-            <div style={{ marginBottom: '7px' }}>
-              <Button color="primary" style={{ marginRight: '7px' }} onClick={() => setInputMode('browse')}>
-                Browse
-              </Button>
-              <Button color="primary" onClick={() => setInputMode('upload')}>
+    <Container maxWidth="sm">
+      <div style={{ marginBottom: '7px' }}>
+        <Button color="primary" style={{ marginRight: '7px' }} onClick={() => setInputMode('browse')}>
+          Browse
+        </Button>
+        <Button color="primary" onClick={() => setInputMode('upload')}>
+          Upload
+        </Button>
+      </div>
+      {inputMode === 'browse' && (
+        <>
+          <Paper className={classes.root}>
+            <InputBase
+              className={classes.input}
+              placeholder="Enter hash e.g. 0773a91efd6547c754fc1d95fb1c62c7d1b47f959c2caa685dfec8736da95c1c"
+              inputProps={{ 'aria-label': 'search swarm nodes' }}
```

Comment:
I find this label a bit misleading, because it's somewhat true that this is what happens on the bee network, however from a user perspective here we want to fetch some content.